File: Data Structures & Algorithms/sort-colors/submission-2.java
class Solution {
    public void sortColors(int[] nums) {
        int[] colors = new int[3];

        for(int i = 0; i < nums.length; i++){
            colors[nums[i]] += 1;
        }

        for(int i = 0; i < nums.length; i++){
            
            for(int j = 0; j < colors.length; j++){
                if (colors[j] != 0){
                    nums[i] = j;
                    colors[j] -= 1;
                    break;
                }
            }

        }

    }
}

File: Data Structures & Algorithms/target-sum/submission-2.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        
        memo = {}
        def helper(index, running_sum):
            nonlocal memo
            if (index, running_sum) in memo:
                return memo[(index, running_sum)]
            if running_sum == target and index == len(nums):
                return 1
            if index == len(nums):
                return 0
            
            memo[(index, running_sum)] = helper(index + 1, running_sum + nums[index]) + helper(index + 1, running_sum - nums[index])
            return memo[(index, running_sum)]
        return helper(0, 0)